README


This archive contains the distribution Dancer-Plugin-Redis,
version 0.7:

  easy database connections for Dancer applications

This software is copyright (c) 2011 by celogeek <me@celogeek.com>.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.




Changes
0.7       2013-01-03 22:50:51 Europe/Paris

    Upgrade test to support new D2 Dancer::Test call

0.6       2013-01-03 00:59:48 Europe/Paris

    Simplify test

0.5       2013-01-02 20:18:07 Europe/Paris

    * use plugin_args instead of dancer_version.

0.4       2012-12-28 20:09:43 Europe/Paris

    * Support of Dancer2

0.3       2012-11-28 19:11:49 Europe/Paris

    * update meta
    * update redis deps : use password generator from it

0.2       2012-07-17 15:28:31 Europe/Paris

    Move to bundle plugin version number

0.14      2012-07-16 22:32:14 Europe/Paris

    Use Dist::Zilla::PluginBundle::Author::Celogeek v0.7
    move RedisWithAuthSupport outside main module

0.13      2012-07-08 18:52:22 Europe/Paris

        Use Dist::Zilla Bundle Celogeek

[celogeek]

0.12    2012-06-08

        Change author name

0.11    2012-06-04

        Fix password connection for multiple connection.

0.10    2012-06-02

        Add support of password option (requirepass)

0.09    2012-06-01

        fix pod

[Ovid / celogeek]

0.08    2012-05-31

        [Ovid]
        Add some basic tests
        Add meta and markdown

        [celogeek]
        Add link to github project

[celogeek]

0.07    2012-05-21

        remove useless deps

0.06    2012-05-19

        simplify the code
        use latest reconnect option of Redis module


0.05    2012-05-12

        change base author
        fix test unit

0.04    2012-04-23

        fix warnings if connection loose

0.03    2011-06-26
        
        Optimize redis connection lost and debug message
        Revamp code
        Remove time ping

[Christophe Nowicki]

0.02    2011-01-01
        Change github url and copyright

0.01    2011-01-01
        Initial version developed based on David Precious Dancer::Plugin::Database



t/lib/TestApp.pm
#
# This file is part of Dancer-Plugin-Redis
#
# This software is copyright (c) 2011 by celogeek <me@celogeek.com>.
#
# This is free software; you can redistribute it and/or modify it under
# the same terms as the Perl 5 programming language system itself.
#
package TestApp;

use Dancer;
use FakeRedis;
use Dancer::Plugin::Redis;

set serializer => 'JSON';

get '/' => sub {
    [ redis->get('foo') ];
};

true;


t/lib/FakeRedis.pm
#
# This file is part of Dancer-Plugin-Redis
#
# This software is copyright (c) 2011 by celogeek <me@celogeek.com>.
#
# This is free software; you can redistribute it and/or modify it under
# the same terms as the Perl 5 programming language system itself.
#
package FakeRedis;

use strict;
use warnings;
{

    package Redis;
    $INC{'Redis.pm'} = 1;
    our $VERSION = 2;
    our $AUTOLOAD;

    sub new {
        my ( $class, %args ) = @_;
        bless \%args => $class;
    }

    sub AUTOLOAD {
        shift;
        my $name = $AUTOLOAD;
        $name =~ s/.*://;
        return $name, @_;
    }
}

1;


lib/Dancer/Plugin/Redis.pm
#
# This file is part of Dancer-Plugin-Redis
#
# This software is copyright (c) 2011 by celogeek <me@celogeek.com>.
#
# This is free software; you can redistribute it and/or modify it under
# the same terms as the Perl 5 programming language system itself.
#
package Dancer::Plugin::Redis;

# ABSTRACT: easy database connections for Dancer applications

use strict;
use warnings;
our $VERSION = '0.7';    # VERSION
use Carp;
use Dancer::Plugin;
use Try::Tiny;
use Redis 1.955;

my $_settings;
my $_handles;

sub redis {
    my @args = @_;
    my ( undef, $name ) = plugin_args(@args);
    $name = "_default" if not defined $name;
    return $_handles->{$name} if exists $_handles->{$name};

    $_settings ||= plugin_setting;

    my $conf
        = $name eq '_default'
        ? $_settings
        : $_settings->{connections}->{$name};
    croak "$name is not defined in your redis conf, please check the doc"
        unless defined $conf;

    return $_handles->{$name} = Redis->new(
        server    => $conf->{server},
        debug     => $conf->{debug},
        encoding  => $conf->{encoding},
        reconnect => $conf->{reconnect} // 60,
        password  => $conf->{password},
    );

}

register redis => \&redis;
register_plugin for_versions => [ 1, 2 ];

1;

__END__

=pod

=head1 NAME

Dancer::Plugin::Redis - easy database connections for Dancer applications

=head1 VERSION

version 0.7

=head1 SYNOPSIS

    use Dancer;
    use Dancer::Plugin::Redis;

    # Calling the redis keyword will get you a connected Redis Database handle:
    get '/widget/view/:id' => sub {
        template 'display_widget', { widget => redis->get('hash_key'); };
    };

    dance;

Redis connection details are read from your Dancer application config - see
below.

=head1 DESCRIPTION

Provides an easy way to obtain a connected Redis database handle by simply calling
the redis keyword within your L<Dancer> application.

=head1 METHODS

=head2 redis

Keywords redis, that use your config to connect to redis

=head1 CONFIGURATION

Connection details will be taken from your Dancer application config file, and
should be specified as, for example: 

    plugins:
        Redis:
            server: '127.0.0.1:6379'
            debug: 0
            encoding: utf8
            reconnect: 60
            password: yourpassword
            connections:
                test:
                    server: '127.0.0.1:6380'
                    debug: 1
                    encoding: utf8
                    password: yourpassword

C<server> is the ip:port of redis server

C<debug> activate the debug of redis

C<encoding> activate auto encoding, if you want raw data, put nothing after encoding

C<reconnect> is the number of second which try to reconnect if we have lost connection, default to 60

C<password> pass AUTH to Redis if you use the requirepass config. You can skip this option if you don't have requirepass set.

=head1 GETTING A DATABASE HANDLE

Calling C<redis> will return a connected database handle; the first time it is
called, the plugin will establish a connection to the database, and return a
reference to the DBI object.  On subsequent calls, the same DBI connection
object will be returned. The connection will be refresh automatically with the Redis C<reconnect> option.

If you have declared named connections as described above in 'DEFINING MULTIPLE
CONNECTIONS', then calling the database() keyword with the name of the
connection as specified in the config file will get you a database handle
connected with those details.

=head1 SEE ALSO

L<Dancer>

L<DBI>

L<Redis>

=head1 BUGS

Please report any bugs or feature requests on the bugtracker website
http://tasks.celogeek.com/projects/perl-modules-dancer-plugin-redis

When submitting a bug or request, please include a test-file or a
patch to an existing test-file that illustrates the bug or desired
feature.

=head1 AUTHOR

celogeek <me@celogeek.com>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2011 by celogeek <me@celogeek.com>.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut


lib/RedisWithAuthSupport.pm
#
# This file is part of Dancer-Plugin-Redis
#
# This software is copyright (c) 2011 by celogeek <me@celogeek.com>.
#
# This is free software; you can redistribute it and/or modify it under
# the same terms as the Perl 5 programming language system itself.
#
package RedisWithAuthSupport;

# ABSTRACT: add support to Redis

use strict;
use warnings;
our $VERSION = '0.2';    # VERSION
use parent 'Redis';

## no critic qw(Subroutines::ProhibitUnusedPrivateSubroutines)

sub new {
    my ( $class, %param ) = @_;

    my $self = $class->SUPER::new(%param);
    $self->{password} = delete $param{password};
    $self->__auth;

    return $self;
}

sub __connect {
    my $self = shift;

    $self->SUPER::__connect;
    $self->__auth;

    return;
}

sub __auth {
    my $self = shift;

    $self->auth( $self->{password} ) if defined $self->{password};

    return;
}

1;

__END__
=pod

=head1 NAME

RedisWithAuthSupport - add support to Redis

=head1 VERSION

version 0.2

=head1 METHODS

=head2 new

Add password option to Redis module

    RedisWithAuthSupport->new(password => 'yourpassword');

=head1 BUGS

Please report any bugs or feature requests on the bugtracker website
https://github.com/celogeek/Dancer-Plugin-Redis/issues

When submitting a bug or request, please include a test-file or a
patch to an existing test-file that illustrates the bug or desired
feature.

=head1 AUTHOR

celogeek <me@celogeek.com>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2011 by celogeek <me@celogeek.com>.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut



Build.PL

use strict;
use warnings;

use Module::Build 0.3601;


my %module_build_args = (
  "build_requires" => {
    "Dancer" => 0,
    "Dancer::Test" => 0,
    "File::Find" => 0,
    "File::Spec" => 0,
    "File::Temp" => 0,
    "JSON" => 0,
    "Module::Build" => "0.3601",
    "Scalar::Util" => 0,
    "Test::More" => "0.94",
    "perl" => "5.004"
  },
  "configure_requires" => {
    "Module::Build" => "0.3601"
  },
  "dist_abstract" => "easy database connections for Dancer applications",
  "dist_author" => [
    "celogeek <me\@celogeek.com>"
  ],
  "dist_name" => "Dancer-Plugin-Redis",
  "dist_version" => "0.7",
  "license" => "perl",
  "module_name" => "Dancer::Plugin::Redis",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "Carp" => 0,
    "Dancer::Plugin" => 0,
    "Redis" => "1.955",
    "Try::Tiny" => 0,
    "perl" => "5.010",
    "strict" => 0,
    "warnings" => 0
  },
  "script_files" => []
);


my $build = Module::Build->new(%module_build_args);

$build->create_build_script;
